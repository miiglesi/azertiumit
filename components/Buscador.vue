<template>
  <div role="group">
    <label for="input-live" class="test">Búsqueda:</label>
    <b-form-input
      id="input-live"
      v-model="word"
      :state="wordState"
      aria-describedby="input-live-help input-live-feedback"
      placeholder="Ingresa nombre de repositorio"
      trim
    ></b-form-input>
    <b-form-invalid-feedback id="input-live-feedback">
      Escribe al menos 3 letras
    </b-form-invalid-feedback>
      <b-form-text id="input-live-help">Repositorios.</b-form-text>
      <!-- Se obtiene el componente sólo si la bandera de resultado es verdadera y le envío un arreglo con los resultados -->
      <TablaResultados :resultados="resultados" v-if="flagResultados"/>
    </div>
</template>
<script>
import TablaResultados from '~/components/TablaResultados.vue'
export default {
  name: 'Buscador',
  components:{
    TablaResultados
  },
  data () {
    return {
      word: '',
      flagResultados:false,
      resultados: {}
    }
  },
  computed: {
    wordState() {
      return this.word.length > 2 ? true : false
    }
  },
  watch: {
    word(change){
      if(this.wordState){
        this.search(change)
      }
    }
  },
  methods: {
    async search(){
      // Creo un arreglo con los parámetros para consumir la API
      const parametros = {
        'word': this.word,
        'per_page': 10,
        'order': 'desc'
      }
      // Hago una petición a la API enviando el arreglo parametros
      const res = await this.$store.dispatch('busqueda/searchword', parametros)
      // Si la longitud de la respuesta de la API es mayor a cero, le asigna a la variable 'resultados' los resultados obtneidos y enciende la bandera
      if(res.items.length>0){
        this.resultados = res.items
        this.flagResultados = true
      } else {
        this.flagResultados = false
      }
    }
  }
}
</script>
<style lang="scss">
@import '~/assets/scss/global';
label {
  color: $darkBlue;
}
</style>

