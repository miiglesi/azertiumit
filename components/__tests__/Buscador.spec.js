import { shallowMount } from "@vue/test-utils";
import Buscador from "../Buscador";


const factory = () => {
  return shallowMount(Buscador, {
  });
};

describe("Buscador", () => {
  test("mounts properly", () => {
    const wrapper = factory();
    expect(wrapper.isVueInstance()).toBeTruthy();
  });

  test("renders properly", () => {
    const wrapper = factory();
    expect(wrapper.html()).toMatchSnapshot();
  });
});