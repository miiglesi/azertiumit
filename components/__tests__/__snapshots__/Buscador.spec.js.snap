// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Buscador renders properly 1`] = `
<div role="group"><label for="input-live" class="test">BÃºsqueda:</label>
  <b-form-input id="input-live" aria-describedby="input-live-help input-live-feedback" placeholder="Ingresa nombre de repositorio" trim=""></b-form-input>
  <b-form-invalid-feedback id="input-live-feedback">
    Escribe al menos 3 letras
  </b-form-invalid-feedback>
  <b-form-text id="input-live-help">Repositorios.</b-form-text>
  <!---->
</div>
`;
